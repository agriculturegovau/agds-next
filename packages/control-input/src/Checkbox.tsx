import React, {
	forwardRef,
	PropsWithRef,
	DetailedHTMLProps,
	InputHTMLAttributes,
} from 'react';
import { CheckboxIndicator } from './CheckboxIndicator';
import { ControlInput } from './ControlInput';
import { ControlContainer } from './ControlContainer';
import { ControlLabel } from './ControlLabel';
import { ControlSize } from './utils';

export type CheckboxProps = PropsWithRef<
	Omit<
		DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>,
		'size'
	> & {
		invalid?: true;
		valid?: true;
		size?: ControlSize;
	}
>;

export const Checkbox = forwardRef<HTMLInputElement, CheckboxProps>(
	function Checkbox(
		{ children, disabled, checked, invalid, valid, size = 'md', ...props },
		ref
	) {
		return (
			<ControlContainer disabled={disabled} valid={valid} invalid={invalid}>
				<ControlInput
					ref={ref}
					type="checkbox"
					checked={checked}
					disabled={disabled}
					{...props}
				/>
				<CheckboxIndicator
					checked={checked}
					disabled={disabled}
					invalid={invalid}
					size={size}
					valid={valid}
				/>
				<ControlLabel disabled={disabled}>{children}</ControlLabel>
			</ControlContainer>
		);
	}
);
