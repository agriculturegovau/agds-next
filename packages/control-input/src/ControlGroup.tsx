import { PropsWithChildren } from 'react';
import { Flex, Stack } from '@ag.ds-next/box';
import { mapSpacing } from '@ag.ds-next/core';
import {
	FieldContainer,
	FieldHint,
	FieldLabel,
	FieldMessage,
} from '@ag.ds-next/field';

export type ControlGroupProps = PropsWithChildren<{
	/** If true, children will be stacked vertically. */
	block?: boolean;
	/** Provides extra information about the field. */
	hint?: string;
	/** Defines an identifier (ID) which must be unique. */
	id?: string;
	/** If true, the invalid state will be rendered. */
	invalid?: boolean;
	/** Describes the purpose of the field. */
	label?: string;
	/** Message to show when the field is invalid. */
	message?: string;
	/** If false, "(optional)" will be appended to the label. */
	required?: boolean;
}>;

export const ControlGroup = ({
	block = false,
	children,
	hint,
	id,
	invalid = false,
	label,
	message,
	required = false,
}: ControlGroupProps) => (
	<FieldContainer invalid={invalid} id={id}>
		<fieldset css={{ padding: 0, margin: 0, border: 'none' }}>
			{label ? (
				<FieldLabel as="legend" required={required}>
					{label}
				</FieldLabel>
			) : null}
			<Stack gap={0.5} css={{ marginTop: label ? mapSpacing(0.5) : undefined }}>
				{hint ? <FieldHint>{hint}</FieldHint> : null}
				{message && invalid ? (
					<FieldMessage invalid={invalid}>{message}</FieldMessage>
				) : null}
				<Flex
					gap={1}
					flexDirection={block ? 'column' : 'row'}
					width="100%"
					paddingTop={0.5}
				>
					{children}
				</Flex>
			</Stack>
		</fieldset>
	</FieldContainer>
);
