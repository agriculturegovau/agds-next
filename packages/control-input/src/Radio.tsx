import { forwardRef, InputHTMLAttributes, PropsWithChildren } from 'react';
import { RadioIndicator } from './RadioIndicator';
import { ControlInput } from './ControlInput';
import { ControlContainer } from './ControlContainer';
import { ControlLabel } from './ControlLabel';
import { ControlSize } from './utils';

type NativeRadioProps = InputHTMLAttributes<HTMLInputElement>;

type BaseCheckboxProps = PropsWithChildren<{
	'aria-required'?: NativeRadioProps['aria-required'];
	disabled?: NativeRadioProps['disabled'];
	checked?: NativeRadioProps['checked'];
	id?: NativeRadioProps['id'];
	max?: NativeRadioProps['max'];
	maxLength?: NativeRadioProps['maxLength'];
	min?: NativeRadioProps['min'];
	minLength?: NativeRadioProps['minLength'];
	name?: NativeRadioProps['name'];
	onBlur?: NativeRadioProps['onBlur'];
	onChange?: NativeRadioProps['onChange'];
	pattern?: NativeRadioProps['pattern'];
	required?: NativeRadioProps['required'];
	value?: NativeRadioProps['value'];
}>;

export type RadioProps = BaseCheckboxProps & {
	/** If true, the invalid state will be rendered. */
	invalid?: boolean;
	/** If true,  the valid state will be rendered. */
	valid?: boolean;
	/** The size of the input. */
	size?: ControlSize;
};

export const Radio = forwardRef<HTMLInputElement, RadioProps>(function Radio(
	{ children, disabled, invalid, valid, size = 'md', ...props },
	ref
) {
	return (
		<ControlContainer disabled={disabled}>
			<ControlInput ref={ref} type="radio" disabled={disabled} {...props} />
			<RadioIndicator
				disabled={disabled}
				invalid={invalid}
				size={size}
				valid={valid}
			/>
			<ControlLabel disabled={disabled}>{children}</ControlLabel>
		</ControlContainer>
	);
});
