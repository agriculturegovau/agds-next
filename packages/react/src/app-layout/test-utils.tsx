import { Fragment } from 'react';
import { Text } from '../text';
import { NotificationBadge } from '../notification-badge';
import { VisuallyHidden } from '../a11y';
import {
	ChartLineIcon,
	EmailIcon,
	ExitIcon,
	HelpIcon,
	HomeIcon,
	SuccessIcon,
	SettingsIcon,
	FactoryIcon,
	AvatarIcon,
	ChevronsLeftIcon,
	MinusIcon,
} from '../icon';
import {
	DropdownMenuDivider,
	DropdownMenuItem,
	DropdownMenuItemLink,
	DropdownMenuPanel,
	DropdownMenuGroup,
	DropdownMenuGroupLink,
	DropdownMenuItemRadio,
} from '../dropdown-menu';

export const navigationItems = (businessName: string) => [
	{
		options: { disableGroupPadding: true },
		items: [
			{
				label: 'Back to my account',
				icon: ChevronsLeftIcon,
				href: '/account',
			},
		],
	},
	{
		options: { disableGroupPadding: true },
		items: [
			{
				label: (
					<Fragment>
						<Text fontWeight="bold" fontSize="xs">
							{businessName}
						</Text>
						<Text color="muted" fontSize="xs">
							ABN: 00 000 000 000
						</Text>
					</Fragment>
				),
			},
		],
	},
	[
		{
			label: 'Dashboard',
			icon: HomeIcon,
			href: '/',
		},
		{
			label: 'Establishments',
			icon: FactoryIcon,
			href: '/establishments',
			items: [
				{
					label: 'Canberra',
					icon: MinusIcon,
					href: '/establishments/canberra',
				},
				{
					label: 'Sydney',
					icon: MinusIcon,
					href: '/establishments/sydney',
				},
			],
		},
		{
			label: 'Data and Insights',
			icon: ChartLineIcon,
			href: '/intelligence',
		},
		{
			label: 'Compliance',
			icon: SuccessIcon,
			href: '/compliance',
			items: [
				{
					label: 'Audit',
					icon: MinusIcon,
					href: '/compliance/audit',
				},
				{
					label: 'Certificates',
					icon: MinusIcon,
					href: '/compliance/certificates',
				},
			],
		},
	],

	[
		{
			label: 'Help',
			icon: HelpIcon,
			href: '/help',
		},
	],
	[
		{
			label: 'Sign out',
			onClick: console.log,
			icon: ExitIcon,
		},
	],
];

export const exampleData = {
	userNames: {
		short: 'Ab Cd',
		regular: 'Toto Wolff',
		medium: 'Benjamin Harrington',
		long: 'Benjamin Alexander Harrington-Smythe',
	},
	businessNames: {
		short: ['Lorem', 'Ipsum', 'Company'],
		regular: ['Antfix', 'Produce Fresh', 'Organic Co'],
		medium: [
			'Malesuada lacus a tortor blandit fermentum phasellus',
			'Iaculis tortor duis ante nec risus elementum id ui',
			'Purus tortor lacus malesuada phasellus ipsum ex duis libero ante id',
		],
		long: [
			'Duis quis iaculis libero nec consectetur luctus ex in malesuada lacus a tortor blandit fermentum phasellus iaculis ipsum nec purus aliquet id elementum risus duis ante ui',
			'Duis ut a libero duis nisl ex lacus in luctus viverra iaculis iaculis tortor duis malesuada blandit purus diam ipsum ante nec risus elementum id ui nec quis aliquet phasellus fermentum',
			'Blandit iaculis iaculis quis ante diam viverra elementum ui risus nec luctus purus tortor lacus malesuada phasellus ipsum ex duis libero ante id Duis fermentum nisl aliquet duis in a nec ut consectetur',
		],
	},
};

export function ExampleAccountDropdown({
	businesses = exampleData.businessNames.regular,
	onBusinessChange,
	selectedBusinessName,
}: {
	businesses?: string[];
	onBusinessChange: (value: string) => void;
	selectedBusinessName: string;
}) {
	return (
		<DropdownMenuPanel palette="light">
			<DropdownMenuGroup label="Businesses">
				{businesses.map((businessName) => (
					<DropdownMenuItemRadio
						key={businessName}
						checked={businessName === selectedBusinessName}
						secondaryText="ABN 00 000 000 000"
						onClick={() => onBusinessChange?.(businessName)}
					>
						{businessName}
					</DropdownMenuItemRadio>
				))}
				<DropdownMenuGroupLink href="#">View all</DropdownMenuGroupLink>
			</DropdownMenuGroup>
			<DropdownMenuDivider />
			<DropdownMenuGroup label="My account">
				<DropdownMenuItemLink href="/profile" icon={AvatarIcon}>
					Profile
				</DropdownMenuItemLink>
				<DropdownMenuItemLink
					href="/messages"
					icon={EmailIcon}
					endElement={
						<span>
							<NotificationBadge tone="action" value={6} max={99} aria-hidden />
							<VisuallyHidden>, 6 unread</VisuallyHidden>
						</span>
					}
				>
					Messages
				</DropdownMenuItemLink>
				<DropdownMenuItemLink href="/account-settings" icon={SettingsIcon}>
					Account settings
				</DropdownMenuItemLink>
			</DropdownMenuGroup>
			<DropdownMenuDivider />
			<DropdownMenuItem onClick={() => console.log('sign out')} icon={ExitIcon}>
				Sign out
			</DropdownMenuItem>
		</DropdownMenuPanel>
	);
}
