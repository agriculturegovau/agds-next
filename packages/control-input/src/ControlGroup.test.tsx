import '@testing-library/jest-dom';
import 'html-validate/jest';
import { render, cleanup } from '../../../test-utils';
import { ControlGroup, ControlGroupProps } from './ControlGroup';
import { Checkbox } from './Checkbox';
import { Radio } from './Radio';

afterEach(cleanup);

function renderCheckboxGroup(props?: Partial<ControlGroupProps>) {
	return render(
		<ControlGroup {...props}>
			<Checkbox>Option A</Checkbox>
			<Checkbox>Option B</Checkbox>
			<Checkbox>Option C</Checkbox>
			<Checkbox>Option D</Checkbox>
		</ControlGroup>
	);
}

function renderRadioGroup(props?: Partial<ControlGroupProps>) {
	return render(
		<ControlGroup {...props}>
			<Radio>Option A</Radio>
			<Radio>Option B</Radio>
			<Radio>Option C</Radio>
			<Radio>Option D</Radio>
		</ControlGroup>
	);
}

describe('ControlGroup ', () => {
	describe('With Checkboxes', () => {
		it('renders correctly', () => {
			const { container } = renderCheckboxGroup({
				label: 'Control group label',
			});
			expect(container).toMatchSnapshot();
		});

		it('renders a valid HTML structure', () => {
			const { container } = renderCheckboxGroup({
				label: 'Control group label',
			});
			expect(container).toHTMLValidate({
				extends: ['html-validate:recommended'],
				rules: {
					// react 18s `useId` break this rule
					'valid-id': 'off',
				},
			});
		});

		it('renders correctly when invalid', () => {
			const { container } = renderCheckboxGroup({
				label: 'Control group label',
				hint: 'Hint text',
				message: 'Error message',
				invalid: true,
			});
			expect(container).toMatchSnapshot();
		});

		it('renders a valid HTML structure when invalid', () => {
			const { container } = renderCheckboxGroup({
				label: 'Control group label',
				hint: 'Hint text',
				message: 'Error message',
				invalid: true,
			});
			expect(container).toHTMLValidate({
				extends: ['html-validate:recommended'],
				rules: {
					// react 18s `useId` break this rule
					'valid-id': 'off',
				},
			});
		});
	});

	describe('With Radios', () => {
		it('renders correctly', () => {
			const { container } = renderRadioGroup({ label: 'Control group label' });
			expect(container).toMatchSnapshot();
		});

		it('renders a valid HTML structure', () => {
			const { container } = renderRadioGroup({ label: 'Control group label' });
			expect(container).toHTMLValidate({
				extends: ['html-validate:recommended'],
				rules: {
					// react 18s `useId` break this rule
					'valid-id': 'off',
				},
			});
		});

		it('renders correctly when invalid', () => {
			const { container } = renderRadioGroup({
				label: 'Control group label',
				hint: 'Hint text',
				message: 'Error message',
				invalid: true,
			});
			expect(container).toMatchSnapshot();
		});

		it('renders a valid HTML structure invalid', () => {
			const { container } = renderRadioGroup({
				label: 'Control group label',
				hint: 'Hint text',
				message: 'Error message',
				invalid: true,
			});
			expect(container).toHTMLValidate({
				extends: ['html-validate:recommended'],
				rules: {
					// react 18s `useId` break this rule
					'valid-id': 'off',
				},
			});
		});
	});
});
