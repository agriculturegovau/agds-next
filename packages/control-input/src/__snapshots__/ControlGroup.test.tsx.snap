// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlGroup  With Checkboxes renders correctly 1`] = `
<div>
  <div
    class="css-1904cz9-boxStyles-FieldContainer"
    id="control-group-:r0:"
  >
    <fieldset
      class="css-faxsk2-ControlGroup"
    >
      <legend
        class="css-n7rdvo-boxStyles"
      >
        <span
          class="css-433uqb-boxStyles-Text"
        >
          Control group label
        </span>
        <span
          class="css-1rpt3h8-boxStyles-Text"
        >
          (optional)
        </span>
      </legend>
      <div
        class="css-1ypnbyi-boxStyles-ControlGroup"
      >
        <div
          class="css-1n3rct3-boxStyles"
        >
          <label
            class="css-kwvgrg-boxStyles-ControlContainer"
          >
            <input
              class="css-poh1fz-ControlInput"
              type="checkbox"
            />
            <span
              class="css-gjetbs-boxStyles-CheckboxIndicator"
            >
              <svg
                aria-hidden="true"
                class="css-umoisd-CheckboxIcon"
                clip-rule="evenodd"
                focusable="false"
                role="img"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 13.4286L8.4 19L21 6"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="3"
                />
              </svg>
            </span>
            <span
              class="css-1g33cu8-boxStyles-Text-ControlLabel"
            >
              Option A
            </span>
          </label>
          <label
            class="css-kwvgrg-boxStyles-ControlContainer"
          >
            <input
              class="css-poh1fz-ControlInput"
              type="checkbox"
            />
            <span
              class="css-gjetbs-boxStyles-CheckboxIndicator"
            >
              <svg
                aria-hidden="true"
                class="css-umoisd-CheckboxIcon"
                clip-rule="evenodd"
                focusable="false"
                role="img"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 13.4286L8.4 19L21 6"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="3"
                />
              </svg>
            </span>
            <span
              class="css-1g33cu8-boxStyles-Text-ControlLabel"
            >
              Option B
            </span>
          </label>
          <label
            class="css-kwvgrg-boxStyles-ControlContainer"
          >
            <input
              class="css-poh1fz-ControlInput"
              type="checkbox"
            />
            <span
              class="css-gjetbs-boxStyles-CheckboxIndicator"
            >
              <svg
                aria-hidden="true"
                class="css-umoisd-CheckboxIcon"
                clip-rule="evenodd"
                focusable="false"
                role="img"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 13.4286L8.4 19L21 6"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="3"
                />
              </svg>
            </span>
            <span
              class="css-1g33cu8-boxStyles-Text-ControlLabel"
            >
              Option C
            </span>
          </label>
          <label
            class="css-kwvgrg-boxStyles-ControlContainer"
          >
            <input
              class="css-poh1fz-ControlInput"
              type="checkbox"
            />
            <span
              class="css-gjetbs-boxStyles-CheckboxIndicator"
            >
              <svg
                aria-hidden="true"
                class="css-umoisd-CheckboxIcon"
                clip-rule="evenodd"
                focusable="false"
                role="img"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 13.4286L8.4 19L21 6"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="3"
                />
              </svg>
            </span>
            <span
              class="css-1g33cu8-boxStyles-Text-ControlLabel"
            >
              Option D
            </span>
          </label>
        </div>
      </div>
    </fieldset>
  </div>
</div>
`;

exports[`ControlGroup  With Checkboxes renders correctly when invalid 1`] = `
<div>
  <div
    class="css-1n67ua0-boxStyles-FieldContainer"
    id="control-group-:r2:"
  >
    <fieldset
      aria-describedby="control-group-:r2:-message control-group-:r2:-hint"
      class="css-faxsk2-ControlGroup"
    >
      <legend
        class="css-n7rdvo-boxStyles"
      >
        <span
          class="css-433uqb-boxStyles-Text"
        >
          Control group label
        </span>
        <span
          class="css-1rpt3h8-boxStyles-Text"
        >
          (optional)
        </span>
      </legend>
      <div
        class="css-1ypnbyi-boxStyles-ControlGroup"
      >
        <span
          class="css-lw0v8u-boxStyles-Text"
          id="control-group-:r2:-hint"
        >
          Hint text
        </span>
        <div
          class="css-1xkbxzi-boxStyles"
        >
          <div
            class="css-1ldvxqr-boxStyles"
          >
            <svg
              aria-hidden="false"
              aria-label="Error"
              class="css-vnv4i8-Icon-FieldMessage"
              clip-rule="evenodd"
              focusable="true"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M16.4661 0.5C16.7332 0.5 16.9893 0.6069 17.1771 0.796862L23.2111 6.89871C23.3962 7.08591 23.5 7.33857 23.5 7.60185V16.3935C23.5 16.6595 23.394 16.9146 23.2055 17.1023L17.0712 23.2087C16.8838 23.3953 16.6301 23.5 16.3657 23.5H7.63889C7.37177 23.5 7.11576 23.3931 6.92792 23.2032L0.789033 16.9968C0.60386 16.8095 0.5 16.5569 0.5 16.2935V7.60646C0.5 7.34045 0.605983 7.08541 0.794505 6.89774L6.92885 0.791284C7.11625 0.604732 7.36991 0.5 7.63434 0.5H16.4661ZM16.7071 7.29289C17.0976 7.68342 17.0976 8.31658 16.7071 8.70711L13.4142 12L16.7071 15.2929C17.0976 15.6834 17.0976 16.3166 16.7071 16.7071C16.3166 17.0976 15.6834 17.0976 15.2929 16.7071L12 13.4142L8.70711 16.7071C8.31658 17.0976 7.68342 17.0976 7.29289 16.7071C6.90237 16.3166 6.90237 15.6834 7.29289 15.2929L10.5858 12L7.29289 8.70711C6.90237 8.31658 6.90237 7.68342 7.29289 7.29289C7.68342 6.90237 8.31658 6.90237 8.70711 7.29289L12 10.5858L15.2929 7.29289C15.6834 6.90237 16.3166 6.90237 16.7071 7.29289Z"
                fill="currentColor"
                fill-rule="evenodd"
                stroke="none"
              />
            </svg>
          </div>
          <span
            class="css-tfsyf8-boxStyles-Text"
            id="control-group-:r2:-message"
          >
            Error message
          </span>
        </div>
        <div
          class="css-1n3rct3-boxStyles"
        >
          <label
            class="css-kwvgrg-boxStyles-ControlContainer"
          >
            <input
              aria-describedby="control-group-:r2:-message"
              aria-invalid="true"
              class="css-poh1fz-ControlInput"
              type="checkbox"
            />
            <span
              class="css-lv3v0c-boxStyles-CheckboxIndicator"
            >
              <svg
                aria-hidden="true"
                class="css-umoisd-CheckboxIcon"
                clip-rule="evenodd"
                focusable="false"
                role="img"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 13.4286L8.4 19L21 6"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="3"
                />
              </svg>
            </span>
            <span
              class="css-1g33cu8-boxStyles-Text-ControlLabel"
            >
              Option A
            </span>
          </label>
          <label
            class="css-kwvgrg-boxStyles-ControlContainer"
          >
            <input
              aria-describedby="control-group-:r2:-message"
              aria-invalid="true"
              class="css-poh1fz-ControlInput"
              type="checkbox"
            />
            <span
              class="css-lv3v0c-boxStyles-CheckboxIndicator"
            >
              <svg
                aria-hidden="true"
                class="css-umoisd-CheckboxIcon"
                clip-rule="evenodd"
                focusable="false"
                role="img"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 13.4286L8.4 19L21 6"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="3"
                />
              </svg>
            </span>
            <span
              class="css-1g33cu8-boxStyles-Text-ControlLabel"
            >
              Option B
            </span>
          </label>
          <label
            class="css-kwvgrg-boxStyles-ControlContainer"
          >
            <input
              aria-describedby="control-group-:r2:-message"
              aria-invalid="true"
              class="css-poh1fz-ControlInput"
              type="checkbox"
            />
            <span
              class="css-lv3v0c-boxStyles-CheckboxIndicator"
            >
              <svg
                aria-hidden="true"
                class="css-umoisd-CheckboxIcon"
                clip-rule="evenodd"
                focusable="false"
                role="img"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 13.4286L8.4 19L21 6"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="3"
                />
              </svg>
            </span>
            <span
              class="css-1g33cu8-boxStyles-Text-ControlLabel"
            >
              Option C
            </span>
          </label>
          <label
            class="css-kwvgrg-boxStyles-ControlContainer"
          >
            <input
              aria-describedby="control-group-:r2:-message"
              aria-invalid="true"
              class="css-poh1fz-ControlInput"
              type="checkbox"
            />
            <span
              class="css-lv3v0c-boxStyles-CheckboxIndicator"
            >
              <svg
                aria-hidden="true"
                class="css-umoisd-CheckboxIcon"
                clip-rule="evenodd"
                focusable="false"
                role="img"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 13.4286L8.4 19L21 6"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="3"
                />
              </svg>
            </span>
            <span
              class="css-1g33cu8-boxStyles-Text-ControlLabel"
            >
              Option D
            </span>
          </label>
        </div>
      </div>
    </fieldset>
  </div>
</div>
`;

exports[`ControlGroup  With Radios renders correctly 1`] = `
<div>
  <div
    class="css-1904cz9-boxStyles-FieldContainer"
    id="control-group-:r4:"
  >
    <fieldset
      class="css-faxsk2-ControlGroup"
    >
      <legend
        class="css-n7rdvo-boxStyles"
      >
        <span
          class="css-433uqb-boxStyles-Text"
        >
          Control group label
        </span>
        <span
          class="css-1rpt3h8-boxStyles-Text"
        >
          (optional)
        </span>
      </legend>
      <div
        class="css-1ypnbyi-boxStyles-ControlGroup"
      >
        <div
          class="css-1n3rct3-boxStyles"
        >
          <label
            class="css-kwvgrg-boxStyles-ControlContainer"
          >
            <input
              class="css-poh1fz-ControlInput"
              type="radio"
            />
            <span
              class="css-rz5sks-boxStyles-RadioIndicator"
            >
              <span
                class="css-1mjlqey-boxStyles-RadioIndicator"
              />
            </span>
            <span
              class="css-1g33cu8-boxStyles-Text-ControlLabel"
            >
              Option A
            </span>
          </label>
          <label
            class="css-kwvgrg-boxStyles-ControlContainer"
          >
            <input
              class="css-poh1fz-ControlInput"
              type="radio"
            />
            <span
              class="css-rz5sks-boxStyles-RadioIndicator"
            >
              <span
                class="css-1mjlqey-boxStyles-RadioIndicator"
              />
            </span>
            <span
              class="css-1g33cu8-boxStyles-Text-ControlLabel"
            >
              Option B
            </span>
          </label>
          <label
            class="css-kwvgrg-boxStyles-ControlContainer"
          >
            <input
              class="css-poh1fz-ControlInput"
              type="radio"
            />
            <span
              class="css-rz5sks-boxStyles-RadioIndicator"
            >
              <span
                class="css-1mjlqey-boxStyles-RadioIndicator"
              />
            </span>
            <span
              class="css-1g33cu8-boxStyles-Text-ControlLabel"
            >
              Option C
            </span>
          </label>
          <label
            class="css-kwvgrg-boxStyles-ControlContainer"
          >
            <input
              class="css-poh1fz-ControlInput"
              type="radio"
            />
            <span
              class="css-rz5sks-boxStyles-RadioIndicator"
            >
              <span
                class="css-1mjlqey-boxStyles-RadioIndicator"
              />
            </span>
            <span
              class="css-1g33cu8-boxStyles-Text-ControlLabel"
            >
              Option D
            </span>
          </label>
        </div>
      </div>
    </fieldset>
  </div>
</div>
`;

exports[`ControlGroup  With Radios renders correctly when invalid 1`] = `
<div>
  <div
    class="css-1n67ua0-boxStyles-FieldContainer"
    id="control-group-:r6:"
  >
    <fieldset
      aria-describedby="control-group-:r6:-message control-group-:r6:-hint"
      class="css-faxsk2-ControlGroup"
    >
      <legend
        class="css-n7rdvo-boxStyles"
      >
        <span
          class="css-433uqb-boxStyles-Text"
        >
          Control group label
        </span>
        <span
          class="css-1rpt3h8-boxStyles-Text"
        >
          (optional)
        </span>
      </legend>
      <div
        class="css-1ypnbyi-boxStyles-ControlGroup"
      >
        <span
          class="css-lw0v8u-boxStyles-Text"
          id="control-group-:r6:-hint"
        >
          Hint text
        </span>
        <div
          class="css-1xkbxzi-boxStyles"
        >
          <div
            class="css-1ldvxqr-boxStyles"
          >
            <svg
              aria-hidden="false"
              aria-label="Error"
              class="css-vnv4i8-Icon-FieldMessage"
              clip-rule="evenodd"
              focusable="true"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M16.4661 0.5C16.7332 0.5 16.9893 0.6069 17.1771 0.796862L23.2111 6.89871C23.3962 7.08591 23.5 7.33857 23.5 7.60185V16.3935C23.5 16.6595 23.394 16.9146 23.2055 17.1023L17.0712 23.2087C16.8838 23.3953 16.6301 23.5 16.3657 23.5H7.63889C7.37177 23.5 7.11576 23.3931 6.92792 23.2032L0.789033 16.9968C0.60386 16.8095 0.5 16.5569 0.5 16.2935V7.60646C0.5 7.34045 0.605983 7.08541 0.794505 6.89774L6.92885 0.791284C7.11625 0.604732 7.36991 0.5 7.63434 0.5H16.4661ZM16.7071 7.29289C17.0976 7.68342 17.0976 8.31658 16.7071 8.70711L13.4142 12L16.7071 15.2929C17.0976 15.6834 17.0976 16.3166 16.7071 16.7071C16.3166 17.0976 15.6834 17.0976 15.2929 16.7071L12 13.4142L8.70711 16.7071C8.31658 17.0976 7.68342 17.0976 7.29289 16.7071C6.90237 16.3166 6.90237 15.6834 7.29289 15.2929L10.5858 12L7.29289 8.70711C6.90237 8.31658 6.90237 7.68342 7.29289 7.29289C7.68342 6.90237 8.31658 6.90237 8.70711 7.29289L12 10.5858L15.2929 7.29289C15.6834 6.90237 16.3166 6.90237 16.7071 7.29289Z"
                fill="currentColor"
                fill-rule="evenodd"
                stroke="none"
              />
            </svg>
          </div>
          <span
            class="css-tfsyf8-boxStyles-Text"
            id="control-group-:r6:-message"
          >
            Error message
          </span>
        </div>
        <div
          class="css-1n3rct3-boxStyles"
        >
          <label
            class="css-kwvgrg-boxStyles-ControlContainer"
          >
            <input
              aria-describedby="control-group-:r6:-message"
              aria-invalid="true"
              class="css-poh1fz-ControlInput"
              type="radio"
            />
            <span
              class="css-zot1ri-boxStyles-RadioIndicator"
            >
              <span
                class="css-1mjlqey-boxStyles-RadioIndicator"
              />
            </span>
            <span
              class="css-1g33cu8-boxStyles-Text-ControlLabel"
            >
              Option A
            </span>
          </label>
          <label
            class="css-kwvgrg-boxStyles-ControlContainer"
          >
            <input
              aria-describedby="control-group-:r6:-message"
              aria-invalid="true"
              class="css-poh1fz-ControlInput"
              type="radio"
            />
            <span
              class="css-zot1ri-boxStyles-RadioIndicator"
            >
              <span
                class="css-1mjlqey-boxStyles-RadioIndicator"
              />
            </span>
            <span
              class="css-1g33cu8-boxStyles-Text-ControlLabel"
            >
              Option B
            </span>
          </label>
          <label
            class="css-kwvgrg-boxStyles-ControlContainer"
          >
            <input
              aria-describedby="control-group-:r6:-message"
              aria-invalid="true"
              class="css-poh1fz-ControlInput"
              type="radio"
            />
            <span
              class="css-zot1ri-boxStyles-RadioIndicator"
            >
              <span
                class="css-1mjlqey-boxStyles-RadioIndicator"
              />
            </span>
            <span
              class="css-1g33cu8-boxStyles-Text-ControlLabel"
            >
              Option C
            </span>
          </label>
          <label
            class="css-kwvgrg-boxStyles-ControlContainer"
          >
            <input
              aria-describedby="control-group-:r6:-message"
              aria-invalid="true"
              class="css-poh1fz-ControlInput"
              type="radio"
            />
            <span
              class="css-zot1ri-boxStyles-RadioIndicator"
            >
              <span
                class="css-1mjlqey-boxStyles-RadioIndicator"
              />
            </span>
            <span
              class="css-1g33cu8-boxStyles-Text-ControlLabel"
            >
              Option D
            </span>
          </label>
        </div>
      </div>
    </fieldset>
  </div>
</div>
`;
